lib/database/migrations/20241127123000-create-gaq-views.js

```diff
@@ -0,0 +1,162 @@
+        UNIX_TIMESTAMP(COALESCE(first_tf_timestamp, time_start))                    AS timestamp
+    FROM global_aggregated_quality_detectors AS gaqd
+    INNER JOIN runs as r
+        ON gaqd.run_number = r.run_number
+`;
+
+const SELECT_RUNS_TO_TIMESTAMPS_FOR_GAQ_PERIODS = `
+    SELECT
+        gaqd.data_pass_id,
+        gaqd.run_number,
+        UNIX_TIMESTAMP(COALESCE(last_tf_timestamp, time_end, NOW(3))) AS ordering_timestamp,
+        UNIX_TIMESTAMP(COALESCE(last_tf_timestamp, time_end))         AS timestamp
+    FROM global_aggregated_quality_detectors AS gaqd
+    INNER JOIN runs as r
+        ON gaqd.run_number = r.run_number
+`;
+
+const SELECT_QCF_EFFECTIVE_PERIODS_FROM_TIMESTAMPS_FOR_GAQ_PERIODS = `
+    SELECT
+        gaqd.data_pass_id,
+        gaqd.run_number,
+        COALESCE(UNIX_TIMESTAMP(qcfep.\`from\`), 0) AS ordering_timestamp,
+        UNIX_TIMESTAMP(qcfep.\`from\`)              AS timestamp
+    FROM quality_control_flag_effective_periods AS qcfep
+        INNER JOIN quality_control_flags AS qcf ON qcf.id = qcfep.flag_id
+        INNER JOIN data_pass_quality_control_flag AS dpqcf ON dpqcf.quality_control_flag_id = qcf.id
+        -- Only flags of detectors which are defined in global_aggregated_quality_detectors
+        -- should be taken into account for calculation of gaq_effective_periods
+        INNER JOIN global_aggregated_quality_detectors AS gaqd
+            ON gaqd.data_pass_id = dpqcf.data_pass_id
+            AND gaqd.run_number = qcf.run_number
+            AND gaqd.detector_id = qcf.detector_id
+`;
+
+const SELECT_QCF_EFFECTIVE_PERIODS_TO_TIMESTAMPS_FOR_GAQ_PERIODS = `
+    SELECT
+        gaqd.data_pass_id,
+        gaqd.run_number,
+        UNIX_TIMESTAMP(COALESCE(qcfep.\`to\`, NOW(3))) AS ordering_timestamp,
+        UNIX_TIMESTAMP(COALESCE(qcfep.\`to\`))         AS timestamp
+    FROM quality_control_flag_effective_periods AS qcfep
+        INNER JOIN quality_control_flags AS qcf ON qcf.id = qcfep.flag_id
+        INNER JOIN data_pass_quality_control_flag AS dpqcf ON dpqcf.quality_control_flag_id = qcf.id
+        -- Only flags of detectors which are defined in global_aggregated_quality_detectors
+        -- should be taken into account for calculation of gaq_effective_periods
+        INNER JOIN global_aggregated_quality_detectors AS gaqd
+            ON gaqd.data_pass_id = dpqcf.data_pass_id
+            AND gaqd.run_number = qcf.run_number
+            AND gaqd.detector_id = qcf.detector_id
+`;
+
+const CREATE_GAQ_PERIODS_TIMESTAMPS_VIEW = `
+CREATE OR REPLACE VIEW gaq_periods_timestamps AS
+    SELECT * FROM (
+        SELECT
+            data_pass_id,
+            run_number,
+            timestamp AS \`from\`,
+            NTH_VALUE(timestamp, 2) OVER (
+                PARTITION BY data_pass_id,
+                run_number
+                ORDER BY ap.ordering_timestamp
+                ROWS BETWEEN CURRENT ROW AND 1 FOLLOWING
+            ) AS \`to\`,
```

Comment:
This kind of function are relatively rarely used (in my experience), it might be useful to put a small comment stating what you are doing here?
Also, maybe you can replace the `NTH_VALUE` and `ROWS BETWEEN...` by the `LAG` function? You might need to reverse the order, not sure